librad/src/id.rs

```diff
@@ -214,8 +284,109 @@ pub trait Entity {
             Err(Error::SignatureVerificationFailed)
         }
     }
+
+    fn check_validity(&self, resolver: &impl Resolver<User>) -> Result<(), Error> {
+        let mut keys = HashSet::<PublicKey>::from_iter(self.keys());
+        let mut users = HashSet::<RadicleUri>::from_iter(self.certifiers());
+
+        for (k, s) in self.signatures() {
+            self.check_signature(k, &s.by, &s.sig, resolver)?;
+            match &s.by {
+                Signatory::OwnedKey => {
+                    keys.remove(k);
+                },
+                Signatory::User(user) => {
+                    users.remove(&user);
+                },
+            }
+        }
+        if keys.len() > 0 || users.len() > 0 {
```

Comment:
Do the quorum rules not apply here? :thinking: I might be remembering the spec wrong.